File: planner-prototype/client/src/app/service/pdf.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {environment} from "../../environments/environment";
import {PdfDocument, Route} from "../model/";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class PDFService {

  constructor(private http: HttpClient) {
  }

  downloadPDF(language: string, routeType: string, route: Route): Observable<PdfDocument> {
    return this.http.post<PdfDocument>(environment.knooppuntUrl + '/pdf/' + routeType + '/' + language, route);
  }
}

File: planner-prototype/server/src/main/java/com/knooppuntnet/service/PdfService.java
package com.knooppuntnet.service;

import com.knooppuntnet.domain.Route;
import com.knooppuntnet.domain.pdf.PdfDocument;
import org.springframework.stereotype.Service;

@Service
public interface PdfService {

	PdfDocument createPdf(String type, String language, Route route);
}
